File: day:66/word_break.py
class Solution:
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        #base case
        if(len(s) == 1):
            if(s in wordDict):
                return True;
            else:
                return False;
            pyt
        queue = [0];
        visited = [0] * len(s)
        while(len(queue) != 0):
            start = queue.pop(0);
            if(visited[start] == 0):
                for end in range(start, len(s)):
                    word = s[start:end + 1]
                    if(word in wordDict):
                        if(end == len(s) - 1):
                            return True;
                        queue.append(end + 1);
            visited[start] = 1;
        
        return False;

File: day:66/wordBreak.java
class Solution {
    public boolean wordBreak(String s, List<String> wordDict) {
        // base case
        if(s.length() == 1) {
        if(wordDict.contains(s)) {
            return true;
        }
        else {
            return false;
        }
    }
    Queue<Integer> queue = new LinkedList<>();
    int[] visited = new int[s.length()];
    queue.add(0);
        
    while(!queue.isEmpty()) {
        int start = queue.remove();
        if(visited[start] == 0) {
           for(int end = start; end < s.length(); end++) {
                String word = s.substring(start, end + 1);
                if(wordDict.contains(word)) {
                    if(end == s.length() - 1) {
                        return true;
                    }
                    queue.add(end + 1);
                }
            } 
        }
        visited[start] = 1;
        
    }
    return false;
    }
}
